src/GenreList.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+
+export default function GenreList({ genres, selectedGenre, onSelect }) {
+  
+   const[Movies,setMovies]=useState([]);
+   const MovieGenre = (movgenre) =>{
```

Comment:
For this component, as you have both genres and selectedGenre as props, no need to have additional state.

You can do

```
let selectedMovies = [];
switch(selectedGenre) {
 case 'Adventure'
 selectedMovies = ['Indiana Jones', 'Jurassic Park', 'Pirates of the Caribbean'];
break;
}
```

and in the view you can simply do
`selectedMovies.map(movie => {(<span>{movie}<span/>)})`

> Try to have the minimum required state. If we can derive some value from other values, it should not be in the state.

Reply:
Resolved